### ISSUE TITLE
Bug: { "env": { "browser": true, "node": false } } not respected

### ISSUE THREAD

**Issue by clshortfuse** [eslint-community/eslint-plugin-n#328]
### Environment

Node version: `v20.12.2`
npm version: `10.5.0`
ESLint version: `8.57.0`
eslint-plugin-n version: `17.10.2`
Operating System: `WSL`/`Windows 11`

### What rule do you want to report?

n/no-unsupported-features/node-builtins

### Link to Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNp9j8tOAzEMRX8lyqat1M5QNkhTIbGo+AGWBKSQuNWUxAl59KFq/p04MwVW7Ox7bd/jK49BtXCW1htoDpF3XDmMie1csOyRaaeyBUzNV4ZweQEDKrkwn5E9W2wE/k5vZZJlA+HEnqd2Tvpiw5e8gWh6TEGVDIcl5SqQMcEBj4J3rHbUfwR3ihBI20kTYXkz0GkgNYUMpA3VKQfOCVBHsl5vs97kfY8dtgGUswVfgxac3DeBQ6HxUn3KPf37lyVdfI0Q3DqdTamnDA3HLXg6g6qHmvUD/DQ+1h6qLLiRCWKaVitg9f81VyPwCsep9/VDs75r7kfkgZCHb2/Qibw=

### What did you expect to happen?

`eslint-plugin-n` rules should not apply when `node:false` or `browser:true` is set.

It is not uncommon for projects to have mixed configurations with different eslint rulesets per folder.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by scagood** [comment]
A few things:

1. The `env` option is only there to set additional global variables (and things that appear like globals).
2. I am curious why you're using the eslint node plugin in a non-node environment?

---

Also, hasn't this option been removed in v9, I believe its been replaced with simply using this plugin?

**Comment by clshortfuse** [comment]
> 2\. I am curious why you're using the eslint node plugin in a non-node environment?

As I said in the first comment it's not uncommon for projects to have multiple environments:

* `/.eslintrc.json` (applies to all)
* `/web/.eslintrc.json` (browser)
* `/server/`  (node)
* `/bin/`  (node)
*

It seems it was deprecated as of v9. Seems odd to break functionality like that, but I guess that means eslint no longer supports multiple environments.

**Comment by clshortfuse** [comment]
If there's no way for eslint to notify plugins what environment it's operating in, that would unfortunately fall upon the plugin itself to detect its environment. 

Something like this would work:

```js
if (typeof process !== 'undefined') && (process.release.name === 'node')
```

I understand the reasoning in terms of "globals" but environments aren't just exclusive to what's inside `globalThis`.

**Comment by scagood** [comment]
In this case, I think you're after an override! 

For example:
https://eslint-online-playground.netlify.app/#eNq9UTtPwzAQ/ivWLX2oTSgLUiskhoqVAYkFF8nY1yrg2MZ20lZV/jt2HiSoVcdmiX3fnb/HncBZniotMMUDy43E5MvBErhWzpOttjl5JELzIkflk58C7fEVJXKv7XgU4dFkRVXfvWaehQmFe/LcXsexPlnBrKb6tHrv0N6ALUEnM+UtDxxaBZYTVYRQ0CVamwl0FJbkPdYIqaH4UdhmsoP6bKbTdBqeobCZ9Z2oytj3NxtrsTsWvS2wq1fDmYNHJYbUTd3IYpeppUotcp0H+wIFha5j0xy6h66o7eK9KNgwG7AX47OQ4Jl05Dl7Q+sCGDEKknl0vlfxz8cF7y33mf1WfPwFI1VYjWH8m+3i8oeL8UdTZ0ch16KQ4VwTUhBYrtHETBTPGrstMYWnZstp9DoU3WoNSmv8Kjhv0p+3vj8WD8niLrlvnAfdFVS/6Vf6ww==

The subsequent versions of eslint require you to specfiy files explicitly matching rules

**Comment by scagood** [comment]
> Something like this would work:
> 
> ```js
> if (typeof process !== 'undefined') && (process.release.name === 'node')
> ```

This would not work because none of the linted code is executed anywhere other than in node.

---

> I understand the reasoning in terms of "globals" but environments aren't just exclusive to what's inside `globalThis`.

You're correct here, it looks like other `languageOptions` are specified (eg I had to add `ecmaVersion: 'latest'` to the example above.

Either way, this can be configured using `overrides` in the old eslint config!

---

Let me know if you need more help!

**Comment by clshortfuse** [comment]
Thanks. It's unfortunate overrides can't "unextend", so you'd have to manually track every folder. It's fine, just unfortunate. In my case, most projects are written in node (build scripts, test, actual library), and then there's sometimes a small UI built for web.

I will see if I bug the eslint team about this, since I have some projects that support Chrome, Firefox, Webkit, Deno, and Node all at once. Even in CSS linting, we split by environment because of limited browser support (I maintain the stylelint `doiuse` project). It just seems sensible that some Javascript linting is environment specific, just like CSS.
